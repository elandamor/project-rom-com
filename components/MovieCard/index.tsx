import { Image, StyleSheet, Text, View } from "react-native";
import { windowWidth } from "../../constants";
import slices from "../../slices";
import { Movie } from "../../types";

const cardWidth = windowWidth / 3 - 15;

export default function MovieCard({ data }: { data: Movie }) {
  const imageSource = slices[data["poster-image"].replace(".jpg", "")];
  const placeholderSource = slices["placeholderPoster"];

  return (
    <View style={styles.container}>
      <Image
        resizeMode="contain"
        source={imageSource ?? placeholderSource}
        style={styles.image}
      />
      <Text
        style={{
          color: "#fff",
          fontFamily: "TitilliumWeb_300Light",
          marginTop: 12,
        }}
        numberOfLines={1}
      >
        {data.name}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-start",
    margin: 5,
    width: cardWidth,
  },
  image: {
    height: cardWidth * 1.5,
    width: cardWidth,
  },
});
