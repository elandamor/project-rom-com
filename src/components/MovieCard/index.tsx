import { Box, Text } from "native-base";
import { Image } from "react-native";
import { windowWidth } from "../../constants";
import slices from "../../slices";
import { Movie } from "../../types";

const cardWidth = windowWidth / 3 - 15;

export default function MovieCard({ data }: { data: Movie }) {
  const imageSource = slices[data["poster-image"].replace(".jpg", "")];
  const placeholderSource = slices["placeholderPoster"];

  return (
    <Box margin={1} width={cardWidth}>
      <Image
        testID="movie-poster"
        resizeMode="contain"
        source={imageSource ?? placeholderSource}
        style={{
          height: cardWidth * 1.5,
          width: cardWidth,
        }}
      />
      <Text testID="movie-name" marginTop={3} numberOfLines={1}>
        {data.name}
      </Text>
    </Box>
  );
}
