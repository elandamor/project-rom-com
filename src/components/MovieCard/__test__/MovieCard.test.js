import { render, screen } from "@testing-library/react-native";
import { NativeBaseProvider } from "native-base";
import React from "react";

import MovieCard from "../index";

const testData = { name: "Rear Window", "poster-image": "poster2.jpg" };
const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

describe("<MovieCard />", () => {
  it("renders correctly", () => {
    render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <MovieCard data={testData} />
      </NativeBaseProvider>
    );

    expect(screen.getByTestId("movie-poster")).toBeTruthy();
    expect(screen.getByTestId("movie-name")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
