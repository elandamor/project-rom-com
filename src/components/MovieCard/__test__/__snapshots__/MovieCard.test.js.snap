// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "marginBottom": 4,
        "marginLeft": 4,
        "marginRight": 4,
        "marginTop": 4,
        "width": 235,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "height": 352.5,
          "width": 235,
        }
      }
      testID="movie-poster"
    />
    <Text
      dataSet={{}}
      numberOfLines={1}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "marginTop": 12,
          "textDecorationLine": undefined,
        }
      }
      testID="movie-name"
    >
      Rear Window
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
